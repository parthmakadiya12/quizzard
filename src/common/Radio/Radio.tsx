import { FC } from "react";
import styled from "styled-components";

interface PropTypes {
  name: string;
  clickHandler: (name: string, queUrl: string, choiceUrl: string) => void;
  choiceUrl: string;
  queUrl: string;
  checked: boolean;
  selected?: string;
}

export const Radio: FC<PropTypes> = ({
  name,
  choiceUrl,
  queUrl,
  clickHandler,
  checked,
  selected,
}) => {
  const currentActive = checked || selected === name;
  return (
    <div>
      <Label>
        <RadioInput
          value={name}
          type="radio"
          name="radio"
          defaultChecked={currentActive}
        />
        <Span
          onClick={() => clickHandler(name, queUrl, choiceUrl)}
          Checked={currentActive}
        >
          {name}
        </Span>
      </Label>
    </div>
  );
};

const Label = styled.div`
  display: flex;
  cursor: pointer;
  font-weight: 500;
  position: relative;
  overflow: hidden;
`;

const RadioInput = styled.input`
  position: absolute;
  left: -9999px;
`;

const Span = styled("span")<{ Checked: boolean }>`
  display: flex;
  align-items: center;
  padding: 0.375em 0.75em 0.375em 0.375em;
  border-radius: 99em;
  transition: 0.25s ease;
  &:before {
    display: flex;
    flex-shrink: 0;
    content: "";
    background-color: #fff;
    width: 1.5em;
    height: 1.5em;
    border-radius: 50%;
    margin-right: 0.375em;
    transition: 0.25s ease;
    box-shadow: ${(props) =>
      props.Checked
        ? `inset 0 0 0 0.4375em ${props.theme.color.title}`
        : `inset 0 0 0 0.125em ${props.theme.color.title}`};
  }
`;
