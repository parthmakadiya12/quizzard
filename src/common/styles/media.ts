const size: any = {
  mobileS: 320,
  mobileM: 375,
  mobileL: 425,
  tablet: 768,
  laptop: 1024,
  laptopL: 1440,
  desktop: 2560,
};
const media: any = Object.keys(size).reduce((acc: any, cur: any) => {
  acc[cur] = `(min-width: ${size[cur]}px)`;
  return acc;
}, {});
export default media;
