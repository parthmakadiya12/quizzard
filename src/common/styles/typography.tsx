import styled from "styled-components";

export const H1 = styled.h1`
  font-weight: 300;
  font-size: 2rem;
`;

export const H2 = styled.h2`
  font-weight: 300;
  font-size: 1.5rem;
`;

export const H3 = styled.h3`
  font-weight: 300;
  font-size: 1.2rem;
`;

export const H4 = styled.h4`
  font-weight: 300;
  font-size: 1rem;
`;

export const P = styled.p`
  font-weight: 300;
  font-size: 0.8rem;
  margin: 0;
  padding: 0;
`;
