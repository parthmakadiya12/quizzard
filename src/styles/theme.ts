import { createGlobalStyle } from "styled-components";

export const darkTheme = {
  color: {
    body: "#262c4a",
    cardBackGround: "#262c4a",
    fontPrimary: "#fdfefe", // white shade
    fontSecondry: "#cec3c3", // dirty white
    headerColor: "#8d95bb", // light purple
    checkBox: "#117fec", // blue
    title: "#8c94bd", // wild blue
    button: "#127eeb", //light blue
    buttonSecondry: "#4098f0", //lighter blue
    buttonFontColor: "white",
    inputColor: "#4a5171",
    inputFontColor: "white",
  },
  type: "dark",
};

export const lightTheme = {
  color: {
    body: "#ffda62",
    cardBackGround: "#ffffff",
    fontPrimary: "#1d345b",
    fontSecondry: "#1A1A1A",
    headerColor: "black",
    checkBox: "#ff5261",
    title: "#1f345a",
    button: "#6287FF",
    buttonSecondry: "#2F60FF",
    buttonFontColor: "white",
    inputColor: "white",
    inputFontColor: "#1d345b",
  },
  type: "light",
};

export const GlobalStyles = createGlobalStyle<{ theme: any }>`
    body{
        background-color: ${({ theme }) => theme.color.body};
        color:${({ theme }) => theme.color.fontPrimary}
    }
`;
