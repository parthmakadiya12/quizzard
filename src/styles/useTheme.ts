import { useState } from 'react'

const useTheme=():[string,()=>void]=> {
    const [theme,setTheme]=useState("dark");

    const toggleTheme=():void=>{
        theme==="dark"? setTheme("light"):setTheme("dark");
    }
    
    return [theme,toggleTheme];
}

export default useTheme;