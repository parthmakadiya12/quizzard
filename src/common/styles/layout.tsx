import styled, { css } from "styled-components";
import media from "./media";

export const Row = styled.div<any>`
  display: flex;
  flex-direction: row;
  ${(props) =>
    props.justifyContent &&
    css`
      justify-content: ${props.justifyContent};
    `}
  ${(props) =>
    props.alignItems &&
    css`
      align-items: ${props.alignItems};
    `}
`;

export const Col = styled.div`
  display: flex;
  flex-direction: column;
`;

export const Wrapper = styled(Col)`
  background-color: ${({ theme }) => theme.color.cardBackGround};
  border: 3px solid #21486b;
  border-radius: 0.8rem;
  padding: 2rem;
  margin: 0.5rem;

  margin: auto;
  ${({ theme }) =>
    theme.type === "light" &&
    `
      border:none;
      box-shadow: rgba(17, 17, 26, 0.05) 0px 4px 16px, rgba(17, 17, 26, 0.05) 0px 8px 32px;
  `}
  @media ${media.tablet} {
    width: 60%;
  }
`;

export const BigQuestionWrapper = styled(Wrapper)`
  margin: 0 10px;
  @media ${media.tablet} {
    max-width: 600px;
    margin: auto;
  }
`;

export const CardTitle = styled.span`
  color: ${({ theme }) => theme.color.headerColor};
  width: 100%;
  font-size: 1.4rem;
  font-weight: 500;
  margin-bottom: 1rem;
  padding-bottom: 1rem;
  background-image: ${({ theme }) =>
    `linear-gradient(to right, ${theme.color.headerColor} 33%, rgba(255,255,255,0) 0%)`};
  background-position: bottom;
  background-size: 0.5rem 0.2rem;
  background-repeat: repeat-x;
`;

export const SubTitle = styled.span`
  font-size: 1rem;
  text-transform: capitalize;
`;

export const SmallText = styled.span`
  font-size: 0.8rem;
  margin-bottom: 0.5rem;
  color: ${({ theme }) => theme.color.fontSecondry};
`;
