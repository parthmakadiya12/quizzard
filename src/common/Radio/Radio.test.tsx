import renderer from "react-test-renderer";
import { ThemeProvider } from "styled-components";

import Radio from "./Radio";
import { darkTheme } from "../../styles/theme";

describe("<Radio>", () => {
  let wrapper, props;

  beforeEach(() => {
    props = {
      name: "Option",
      choiceUrl: "/questions/1212/choices/12",
      queUrl: "/questions/12",
      clickHandler: jest.fn(),
      checked: false,
      selected: false,
    };
  });
  it("should render correctly", () => {
    wrapper = renderer
      .create(
        <ThemeProvider theme={darkTheme}>
          <Radio {...props} />
        </ThemeProvider>
      )
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it("should call click handler by cling on radio", () => {
    wrapper = mount(
      <ThemeProvider theme={darkTheme}>
        <Radio {...props} />
      </ThemeProvider>
    );

    wrapper.find("span").simulate("click");
    expect(props.clickHandler).toHaveBeenCalled();
  });
});
