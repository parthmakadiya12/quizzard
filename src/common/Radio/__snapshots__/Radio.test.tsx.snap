// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Radio> should render correctly 1`] = `
<div>
  <div
    className="sc-bdnxRM kLuAZG"
  >
    <input
      className="sc-gtsrHT fWHZtE"
      defaultChecked={false}
      name="radio"
      type="radio"
      value="Option"
    />
    <span
      className="sc-dlnjwi CWyZM"
      onClick={[Function]}
    >
      Option
    </span>
  </div>
</div>
`;
